packages/nextra-theme-docs/src/anchor.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ComponentProps, ReactElement } from 'react'
+// eslint-disable-next-line no-restricted-imports -- only in this file we determine either we include <a /> as child of <NextLink /> based of `newNextLinkBehavior` value
+import NextLink, { LinkProps } from 'next/link'
+import { useConfig } from './config'
+
+export default function Anchor({
+  href = '',
+  children,
+  ...props
+}: Omit<ComponentProps<'a'>, 'ref'> &
+  // Use `href` and `onClick` types from <a />
+  Omit<LinkProps, 'href' | 'onClick'>): ReactElement | null {
+  const config = useConfig()
+
+  // When href is '' there is client error in console
+  // Warning: Prop `href` did not match. Server: … Client: …
+  if (!href) {
+    return <a {...props}>{children}</a>
+  }
+
+  const isExternal = href.startsWith('https://')
+  if (isExternal) {
+    return (
+      <a target="_blank" rel="noreferrer" href={href} {...props}>
+        {children}
+      </a>
+    )
+  }
+
+  if (config.newNextLinkBehavior) {
+    return (
+      <NextLink href={href} {...props}>
+        {children}
+      </NextLink>
+    )
+  }
+
+  return (
+    <NextLink href={href} passHref>
+      <a {...props}>{children}</a>
+    </NextLink>
+  )
```

Comment:
generic Anchor component that renders either `<NextLink />` with or without `<a />` tag based on next config setting `experimental.newNextLinkBehavior`